### PR TITLE
Kaleem neslit.9502.card style

### PR DESCRIPTION
**Pull request by KaleemNeslit**
## Link to Issue
Closes: #9502 

## Description of Changes
- use the VirtuosoGrid to show the 3 card in the row 
- in mobile show one card 
- added the cardView in dropdown menu  

## "How We Fixed It"

- use the VirtuosoGrid to show the 3 card in the row 
- in mobile show one card 
- added the cardView in dropdown menu  
- 
## Test Plan
Goto any community page and from the drop down select cardView 
you will see 3 cards in the row and on mobile it will be 1 card

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/AuthorAndPublishInfo/AuthorAndPublishInfo.tsx [hunk 1]
```diff
@@ -85,6 +86,7 @@ export const AuthorAndPublishInfo = ({
   versionHistory,
   activeThreadVersionId,
   onChangeVersionHistoryNumber,
+  isCardView = false,
```
[the remark this answers is not in the document]
Comment (KaleemNeslit) [reply packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/AuthorAndPublishInfo/AuthorAndPublishInfo.tsx]: done

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/markdown_formatted_text.tsx
[the remark this answers is not in the document]
Comment (KaleemNeslit) [reply packages/commonwealth/client/scripts/views/components/react_quill_editor/markdown_formatted_text.tsx:1]: issue is that mostly thread data rendering happing in quil_editior  for move that stuff for card view can b tricky @mzparacha

File: packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadCard.scss [hunk 2]
```diff
@@ -264,3 +265,68 @@
...
   margin-top: 0 !important;
 }
+
+.virtuoso-grid-list {
+  border: 1px solid #f0eff0 !important;
+  gap: 5px !important;
+  display: flex !important;
+  flex-wrap: wrap !important;
+}
+.virtuoso-grid-item {
```
[the remark this answers is not in the document]
Comment (KaleemNeslit) [reply packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadCard.scss:274]: basically this class "virtuoso-grid-item" is from VirtuosoGrid we are just overwriting some of their properties

**Comment by KaleemNeslit** [comment]
https://github.com/user-attachments/assets/3911c9bc-2082-458a-ad81-c2ce67f68b86

@masvelio @mzparacha  here is the update ui video

**Comment by mzparacha** [comment]
1. Cards feel a bit off-paded on mobile (a lot of padding and height)

image

2. This feels a bit congested, can we make it 2 cards per row on certain breakpoints?

image

File: packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadCard.scss [hunk 3]
```diff
@@ -264,3 +265,124 @@
...
+    .content-wrapper
+    .content-body-wrapper
+    .content-body
+    .card-view
+    .card-image-container
+    img {
+    object-fit: contain;
+    max-width: 100%;
+    max-height: 100%;
+  }
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadCard.scss:359]: please nest these in the first .ThreadCard instances above

File: packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadCard.scss [hunk 4]
```diff
@@ -264,3 +265,124 @@
...
+    .card-view
+    .card-image-container {
+    width: 70px;
+    height: 80px;
+    margin-top: 20px;
+  }
+
+  .ThreadCard {
+    height: 300px !important;
+  }
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadCard.scss:387]: same as above

**Comment by masvelio** [comment]
let's make sure to resolve the comments and merge it!

**Comment by masvelio** [comment]
@KaleemNeslit

**Comment by KaleemNeslit** [comment]
> @KaleemNeslit

@mzparacha  i am working with @sssssabinaaa  she need to approve some ui changes then i will update the PR

**Comment by dillchen** [comment]
@KaleemNeslit Can u show the mobile view and second show some image only posts? I want to see what they look like

**Comment by KaleemNeslit** [comment]
> @KaleemNeslit Can u show the mobile view and second show some image only posts? I want to see what they look like

@dillchen  this is how it's looking on different views
https://github.com/user-attachments/assets/08bdb240-7fff-42d7-be34-8851762cf0ec

**Comment by KaleemNeslit** [comment]
@mzparacha  @masvelio  @Israellund  this pr is now available to merge designed changes are approve by @dillchen  @sssssabinaaa

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/markdown_formatted_text.tsx [hunk 5]
```diff
@@ -193,16 +195,31 @@ export const MarkdownFormattedText = ({
...
-        ref={containerRef}
-        className={getClasses<{ collapsed?: boolean }>(
-          { collapsed: isTruncated },
-          customClass || 'MarkdownFormattedText',
-        )}
-      >
-        {finalDoc}
-      </div>
+      {isCardView ? (
+        <div className="card-view">
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/react_quill_editor/markdown_formatted_text.tsx:199]: Seems like we are just adding an extra wrapper element here. Plz move the `isCardView` logic outside of the editor components.

**Review by mzparacha — APPROVED** [review]
Only 1 change requested above. Approving in advance.

**Review by masvelio — APPROVED** [review]
I would like to see one change, approving in advance tho

**Comment by KaleemNeslit** [comment]
@masvelio @mzparacha  i address the last comments .

**Pull request merged**